Ponca/src/Fitting/basket.h

```diff
@@ -15,14 +15,80 @@
 namespace Ponca
 {
 
+#define BSKW typename BasketType::WeightFunction
+#define BSKP typename BasketType::DataPoint
+
 #ifndef PARSED_WITH_DOXYGEN
 /*! \brief Namespace used for structure or classes used internally by the lib */
 namespace internal
 {
+    template <class P, class W,
+        typename Aggregate,
+        template <class, class, typename> class Ext,
+        template <class, class, typename> class... Exts>
+    struct BasketAggregateImpl
+    {
+        using type = typename BasketAggregateImpl<P, W, Ext<P, W, Aggregate>, Exts...>::type;
+    };
+
+    template <class P, class W,
+        typename Aggregate,
+        template <class, class, typename> class Ext>
+    struct BasketAggregateImpl<P, W, Aggregate, Ext>
+    {
+        using type = Ext<P, W, Aggregate>;
+    };
+
     /*! \brief Internal class used to build the Basket structure */
-    template <class, class, typename T> class Forward: public T {};
+    template <class P, class W,
+        template <class, class, typename> class... Exts>
+    struct BasketAggregate : BasketAggregateImpl<P, W, PrimitiveBase<P, W>, Exts...>
```

Comment:
What is the purpose of this intermediate struct since the class `Basket` could just inherits directly from `internal::BasketAggregateImpl<P, W, PrimitiveBase<P, W>, Exts...>::type`?

Reply:
I chose to hide the `PrimitiveBase<P, W>` part from the `Basket` class definition as I thought it could be considered an implementation detail, but it is true that what you mentioned is equivalent (I also thought it made the `Basket` class a bit more readable)